compiler/src/main/scala/org/scalajs/core/compiler/Compat210Component.scala

```diff
@@ -58,6 +66,25 @@ trait Compat210Component {
     }
   }
 
+  // Impl classes disappeared in 2.12.0-M4
+
+  lazy val scalaUsesImplClasses: Boolean =
+    definitions.SeqClass.implClass != NoSymbol // a trait we know has an impl class
+
+  implicit final class StdTermNamesCompat(self: global.nme.type) {
+    val IMPL_CLASS_SUFFIX = "$class"
```

Comment:
Although that line is protected by `isImplClass`, so in fact it would be fine. So again, it all depends on the decision about https://github.com/scala-js/scala-js/pull/2288/files#r55755629